app/process_proposal.go

```diff
@@ -58,10 +58,9 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) (resp abci.Resp
 
 		sdkTx, err := app.txConfig.TxDecoder()(tx)
 		if err != nil {
-			// we don't reject the block here because it is not a block validity
-			// rule that all transactions included in the block data are
-			// decodable
-			continue
+			// An error here means that a tx was included in the block that is not decodable.
+			logInvalidPropBlock(app.Logger(), req.Header, fmt.Sprintf("tx %d is not decodable", idx))
+			return reject()
```

Comment:
FYI: This needs to be versioned

Reply:
https://github.com/celestiaorg/celestia-app/pull/3006/commits/0b7ba89012a26d1add1674e1a4e051409f3d1156